### ISSUE TITLE
Discriminator cond_dim if statement being bypassed

### ISSUE THREAD

**Issue by ajhic2001** [ratschlab/RGAN#13]
This is just a heads up to the authors or anyone who is using conditional inputs, but in the current state of things this if statement is being ignored.
model.py line 275
 
        if cond_dim > 0:
            assert not c is None
            repeated_encoding = tf.stack([c]*seq_length, axis=1)
            inputs = tf.concat([x, repeated_encoding], axis=2)
        else:
            inputs = x

As at model.py line 292, inputs is set to x, rather than being set to inputs.

        rnn_outputs, rnn_states = tf.nn.dynamic_rnn(
            cell=cell,
            dtype=tf.float32,
            inputs=x)

Making this small change allowed me to finally train my first conditional model.

**Comment by corcra** [comment]
Oh wow, thank you so much! I think this may be the bug I have been failing to find for ages... I will update the code right away.

Also, in case you/anyone is worried: we didn't actually use this code for the _conditional_ experiments in the paper, because it didn't work (for reasons I didn't know, until now, hopefully), we were using a different (messier, but seemingly less buggy) version of the code written by @cresteban.

**Comment by ajhic2001** [comment]
No worries, Happy to be able to help. 
I'm not sure if its possible to use batch mean and conditional inputs at the same time, but if so, then X in the batch_means if statement would need to be changed to inputs as defined by the cond_dim if statement.

**Comment by dmortem** [comment]
@corcra  
Hi, what are the differences on the codes between this version and the version for the conditional experiments in the paper?
Thank you!
